### PR TITLE
[SYCL-MLIR][NFC] Fix warnings in polygeist and mlir-sycl

### PR DESCRIPTION
**Pull request by victor-eds**
Fix pipeline warnings (GCC).

Signed-off-by: Victor Perez <victor.perez@codeplay.com>

### CODE REVIEW

File: mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp [hunk 1]
```diff
@@ -26,6 +26,8 @@ llvm::StringRef mlir::sycl::memoryAccessModeAsString(
     return "discard_read_write";
   case MemoryAccessMode::Atomic:
     return "atomic";
+  default:
```
Comment (whitneywhtsang) [inline mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp:29]: As the switch already covered all possible MemoryAccessMode
```
enum class MemoryAccessMode {
  Read = 1024,
  Write,
  ReadWrite,
  DiscardWrite,
  DiscardReadWrite,
  Atomic
};
```
would adding a default case add a warning when compiling with clang++?

Comment (etiotto) [reply mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp:29]: I recall it did in one of my past PRs, and newer version of gcc should warn as well AFAIK.

Comment (etiotto) [reply mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp:29]: I think clang will correctly recognize the llvm_unreachable and suppress the warnings.

**Review by etiotto — APPROVED** [review]

File: mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp:29]: I tried to compile with clang++ and got:
```
llvm/mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp:28:3: warning: default label in switch which covers all enumeration values [-Wcovered-switch-default]
  default:
  ^
1 warning generated.
```

Comment (victor-eds) [reply mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp]: @whitneywhtsang, please, try with last push

**Review by whitneywhtsang — APPROVED** [review]

File: mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp]: I tried just for this switch, and no warnings with the change from the last push.

**Pull request merged**